azerothcore/azerothcore-wotlk
# feat(Core/CreatureAI): improve npc position during the combat

Changed the creature's positions while attacking the player.

Authors: 
@Helias 
@Yehonal
@lucaAngrisani

PR inspired by:
@Langerz82 ( https://gist.github.com/callmephil/5c5d18e1db0ca4e86115a299b836cabf )

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3357

## EXPECTED BLIZZLIKE BEHAVIOR
The first four videos of this playlist
https://www.youtube.com/playlist?list=PLx1yhYXFOxyFCrLWJIr2dfyySx5jiEr4k

## HOW THIS PR WORK
last video of the playlist
https://www.youtube.com/watch?v=ogmPCycMC3U&list=PLx1yhYXFOxyFCrLWJIr2dfyySx5jiEr4k&index=5

## HOW TO TEST
- Attack a group of NPCs and see how they reposition to attack you.
- Later, go really close to the NPCs they should go back a bit to attack you

Be careful:
- inside the dungeons, there is more "tolerance" about reposition, so they can also not reposition instantly
- outside the dungeons, it's easy that the NPCs reposition themselves.

## Target branch(es):

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
